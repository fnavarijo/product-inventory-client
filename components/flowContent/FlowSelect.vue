<template>
  <select class="border border-light-gray-500 bg-white p-5">
    <slot />
  </select>
</template>

<script lang="ts">
// TODO: Can I validate children?
import Vue from 'vue';

export default Vue.extend({
  name: 'FlowSelect',
});
</script>
