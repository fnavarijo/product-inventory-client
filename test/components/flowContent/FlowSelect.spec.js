import { mount } from '@vue/test-utils';

import FlowSelect from '@/components/flowContent/FlowSelect.vue';

describe('FlowSelect component', () => {
  xit('should set the v-model value in the select', async () => {
    const wrapper = mount(FlowSelect, {
      slots: [
        '<option value="desc">Descending</option>',
        '<option value="asc">Ascending</option>',
      ],
    });

    const selectInput = wrapper.find('select');
    await selectInput.setValue('asc');

    const selectedValue = selectInput.element.nodeValue;

    expect(selectedValue).toBe('asc');
  });

  it('should match snapshot', () => {
    const wrapper = mount(FlowSelect, {
      slots: [
        '<option value="desc">Descending</option>',
        '<option value="asc">Ascending</option>',
      ],
    });

    expect(wrapper.html()).toMatchSnapshot();
  });
});
