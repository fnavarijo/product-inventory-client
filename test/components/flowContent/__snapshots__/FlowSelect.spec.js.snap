// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FlowSelect component should match snapshot 1`] = `<select class="border border-light-gray-500 bg-white p-5"></select>`;
